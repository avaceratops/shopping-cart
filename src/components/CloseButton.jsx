export default function CloseButton({ onClick }) {
  return (
    <button
      type="button"
      className="relative -m-2 p-2 text-gray-400 hover:text-gray-500"
      onClick={onClick}
    >
      <span className="sr-only">Close panel</span>
      <svg
        className="h-6 w-6"
        fill="none"
        viewBox="0 0 24 24"
        strokeWidth="1.5"
        stroke="currentColor"
        aria-hidden="true"
      >
        <path strokeLinecap="round" strokeLinejoin="round" d="M6 18L18 6M6 6l12 12"></path>
      </svg>
    </button>
  );
}
