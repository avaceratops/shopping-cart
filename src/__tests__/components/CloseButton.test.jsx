import { vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CloseButton from '../../components/CloseButton';

it('renders button correctly', () => {
  const { container } = render(<CloseButton />);
  expect(container).toMatchSnapshot();
});

it('calls onClick handler when user clicks the button', async () => {
  const onClick = vi.fn();
  const user = userEvent.setup();
  render(<CloseButton onClick={onClick} />);

  const button = screen.getByRole('button', { name: /close panel/i });
  await user.click(button);

  expect(onClick).toHaveBeenCalledTimes(1);
});
