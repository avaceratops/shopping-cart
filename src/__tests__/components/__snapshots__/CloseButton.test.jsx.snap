// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders button correctly 1`] = `
<div>
  <button
    class="relative -m-2 rounded-md p-2 text-gray-400 hover:bg-gray-400 hover:text-white"
    type="button"
  >
    <span
      class="sr-only"
    >
      Close panel
    </span>
    <svg
      aria-hidden="true"
      class="block h-6 w-6"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 18 18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</div>
`;
